control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3743,17 +3744,66 @@ func (r *HostedControlPlaneReconciler) reconcileOperatorLifecycleManager(ctx con
 				return fmt.Errorf("failed to reconcile catalogs: %w", err)
 			}
 
-			catalogsImageStream := manifests.CatalogsImageStream(hcp.Namespace)
-			if !overrideImages {
-				isImageRegistryOverrides := util.ConvertImageRegistryOverrideStringToMap(p.OLMCatalogsISRegistryOverridesAnnotation)
-				if _, err := createOrUpdate(ctx, r, catalogsImageStream, func() error {
-					return olm.ReconcileCatalogsImageStream(catalogsImageStream, p.OwnerRef, isImageRegistryOverrides)
-				}); err != nil {
-					return fmt.Errorf("failed to reconcile catalogs image stream: %w", err)
+			isImageRegistryOverrides := util.ConvertImageRegistryOverrideStringToMap(p.OLMCatalogsISRegistryOverridesAnnotation)
+
+			if r.ManagementClusterCapabilities.Has(capabilities.CapabilityImageStream) {
+				catalogsImageStream := manifests.CatalogsImageStream(hcp.Namespace)
+				if !overrideImages {
+					if _, err := createOrUpdate(ctx, r, catalogsImageStream, func() error {
+						return olm.ReconcileCatalogsImageStream(catalogsImageStream, p.OwnerRef, isImageRegistryOverrides)
+					}); err != nil {
+						return fmt.Errorf("failed to reconcile catalogs image stream: %w", err)
+					}
+				} else {
+					if _, err := util.DeleteIfNeeded(ctx, r, catalogsImageStream); err != nil {
+						return fmt.Errorf("failed to remove OLM Catalog ImageStream: %w", err)
+					}
 				}
-			} else if r.ManagementClusterCapabilities.Has(capabilities.CapabilityImageStream) {
-				if _, err := util.DeleteIfNeeded(ctx, r, catalogsImageStream); err != nil {
-					return fmt.Errorf("failed to remove OLM Catalog ImageStream: %w", err)
+			} else {
+				pullSecret := common.PullSecret(hcp.Namespace)
+				if err := r.Get(ctx, client.ObjectKeyFromObject(pullSecret), pullSecret); err != nil {
+					return fmt.Errorf("failed to get pull secret for namespace %s: %w", hcp.Namespace, err)
+				}
+
+				for name, catalog := range olm.CatalogToImage {
+					parts := strings.Split(catalog, ":")
+					if len(parts) != 2 {
+						return fmt.Errorf("invalid catalog format %s, expected 'registry:tag'", catalog)
+					}
+					catalogImage, tag := parts[0], parts[1]
+
+					catalogImageParts := strings.Split(catalogImage, "/")
+					catalogRepo := catalogImageParts[len(catalogImageParts)-1]
+					catalogRegistry := strings.Join(catalogImageParts[:len(catalogImageParts)-1], "/")
```

Comment:
Please use https://github.com/openshift/hypershift/blob/a11f0a1b12da74f345953f9441221e4418c314f4/support/thirdparty/library-go/pkg/image/reference/reference.go#L32 to parse the image ref

Reply:
and in general use the funcs there to build new refs